src/test/java/apitests/DriverTest.java

```diff
@@ -0,0 +1,195 @@
+
+	List<Driver> drivers;
+
+	Driver driver1;
+	Driver driver2;
+	Driver driver3;
+
+	private MediaType contentType = new MediaType(MediaType.APPLICATION_JSON.getType(),
+			MediaType.APPLICATION_JSON.getSubtype(), Charset.forName("utf8"));
+
+	@Before
+	public void setup() throws Exception {
+
+		mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+		this.addressList.add(addressManager.add(TestModelsFactory.createTestAddress1()));
+		driver1 = driverManager.add(TestModelsFactory.createTestDriver1(addressList.get(0)));
+		driverList.add(driver1);
+		driver2 = driverManager.add(TestModelsFactory.createTestDriver2(addressList.get(0)));
+		driverList.add(driver2);
+		drivers = driverManager.getAll();
+
+	}
+
+	@After
+	public void cleanup() throws Exception {
+
+		for (Driver driverList : driverList) {
+			driverManager.delete(driverList);
+		}
+	}
+
+	@Test
+	public void getDriver() throws Exception {
+		long LongID = driverList.get(1).getId();
+		int IntID = (int) LongID;
+		
+		mockMvc.perform(get("/api/drivers/" + driverList.get(1).getId()))
+				.andExpect(status().isOk())
+				.andExpect(content().contentType(contentType))
+				.andExpect(jsonPath("$.id", is(IntID)))
+				.andExpect(jsonPath("$.firstName", is(driverList.get(1).getFirstName())))
+				.andExpect(jsonPath("$.lastName", is(driverList.get(1).getLastName())))
+				.andExpect(jsonPath("$.salary", is(driverList.get(1).getSalary().doubleValue())))
+				.andExpect(jsonPath("$.pesel", is(driverList.get(1).getPESEL())))
+				.andExpect(jsonPath("$.available", is(driverList.get(1).isAvailable())))
+				.andExpect(jsonPath("$.deleted", is(driverList.get(1).isDeleted())));
+	}
+
+	@Test
+	public void postDriver() throws Exception {
+
+		long LongID = driver2.getId() + 1;
+		int IntID = (int) LongID;
+
+		driver3 = TestModelsFactory.createTestDriver3(addressList.get(0));
+		Driver json = driver3;
+		mockMvc.perform(post("/api/drivers/")
+				.contentType(contentType).content(convertObjectToJsonBytes(json)))
+				.andExpect(status().isOk())
+				.andExpect(content().contentType(contentType))
+				.andExpect(jsonPath("$.id", is(IntID)))
+				.andExpect(jsonPath("$.firstName", is(driver3.getFirstName())))
+				.andExpect(jsonPath("$.lastName", is(driver3.getLastName())))
+				.andExpect(jsonPath("$.salary", is(driver3.getSalary().doubleValue())))
+				.andExpect(jsonPath("$.pesel", is(driver3.getPESEL())))
+				.andExpect(jsonPath("$.available", is(driver3.isAvailable())))
+				.andExpect(jsonPath("$.deleted", is(driver3.isDeleted())));
+
+		driverManager.delete(driver2.getId() + 1);
+	}
+
+	@Test
+	public void deleteDriver() throws Exception {
+
+		driver3 = driverManager.add(TestModelsFactory.createTestDriver3(addressList.get(0)));
+
+		mockMvc.perform(
+				MockMvcRequestBuilders.delete("/api/drivers/" + driver3.getId())
+				.accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk());
+
+		Assert.assertFalse(drivers.contains(driver3));
```

Comment:
Since `drivers` list is populated in `setup()` method, it will never contain `driver3`. You should add driver with driverManager, check if he was added ( `Assert.assertTrue(driverManager.getAll().contains(driver3));`), delete driver with API call and check if he was deleted (`Assert.assertFalse(driverManager.getAll().contains(driver3));`).